js/ui/tree_view/ui.tree_view.base.js

```diff
@@ -54,6 +54,13 @@ var TreeViewBase = HierarchicalCollectionWidget.inherit({
             e.target = $itemElement;
             e.currentTarget = $itemElement;
             this._itemClickHandler(e, $itemElement.children("." + ITEM_CLASS));
+
+            var expandEventName = this._getEventNameByOption(this.option("expandEvent")),
```

Comment:
Should we make it public in 18.1?

Reply:
We may publish this option in 18.1 but in the separate pull request